pkg/rancher-desktop/pages/containers/ContainerInfo.vue

```diff
@@ -0,0 +1,385 @@
+
+// Computed properties
+const containerId = computed(() => route.params.id as string || '');
+
+const currentContainer = computed((): Container | null => {
+  if (!containers.value || !containerId.value) {
+    return null;
+  }
+  return containers.value[containerId.value] || null;
+});
+
+const containerName = computed(() => {
+  if (!currentContainer.value) {
+    return containerId.value.substring(0, 12);
+  }
+  const name = currentContainer.value.containerName;
+  return name.replace(/^\//, '').replace(/_[a-z0-9-]{36}_[0-9]+/, '') || containerId.value.substring(0, 12);
+});
+
+const containerState = computed(() => {
+  if (!currentContainer.value) {
+    return 'unknown';
+  }
+  return currentContainer.value.state || currentContainer.value.status || 'unknown';
+});
+
+const isRunning = computed(() => {
+  if (!currentContainer.value) {
+    return false;
+  }
+  return currentContainer.value.state === 'running' || currentContainer.value.status === 'Up';
+});
+
+// Watchers
+watch(containerName, (name) => {
+  store.dispatch('page/setHeader', {
+    title:       name || 'Container Info',
+    description: '',
+    action:      'ContainerStatusBadge',
+  });
+}, { immediate: true });
+
+// Methods as functions
+const subscribe = async() => {
+  if (subscribeTimer.value) {
+    clearTimeout(subscribeTimer.value);
+  }
+  try {
+    if (!window.ddClient || !isK8sReady.value || !settings.value) {
+      subscribeTimer.value = setTimeout(() => subscribe(), 1_000);
+      return;
+    }
+    await store.dispatch('container-engine/subscribe', {
+      type:   'containers',
+      client: window.ddClient,
+    });
+  } catch (error) {
+    console.error('There was a problem subscribing to container events:', { error });
+  }
+};
+
+const onSearchInput = () => {
+  containerLogs.value?.performSearch(searchTerm.value);
+};
+
+const searchNext = () => {
+  containerLogs.value?.searchNext(searchTerm.value);
+};
+
+const searchPrevious = () => {
+  containerLogs.value?.searchPrevious(searchTerm.value);
+};
+
+const clearSearch = () => {
+  searchTerm.value = '';
+  containerLogs.value?.clearSearch();
+  nextTick(() => {
+    searchInput.value?.focus();
+  });
+};
+
+const handleSearchKeydown = (event: KeyboardEvent) => {
+  if (event.key === 'Enter') {
+    if (event.shiftKey) {
+      searchPrevious();
+    } else {
+      searchNext();
+    }
+    event.preventDefault();
+  } else if (event.key === 'Escape') {
+    clearSearch();
+    event.preventDefault();
+  }
+};
+
+const handleGlobalKeydown = (event: KeyboardEvent) => {
+  if (event.key === '/') {
+    // Don't trigger if search input is already focused
+    if (document.activeElement !== searchInput.value) {
+      event.preventDefault();
+      searchInput.value?.focus();
+      searchInput.value?.select();
+    }
+  }
+};
+
+// Lifecycle hooks
+onMounted(() => {
+  ipcRenderer.on('settings-read', (_event: any, settingsData: any) => {
```

Comment:
```suggestion
  ipcRenderer.on('settings-read', (_event, settingsData) => {
```
`ipcRenderer.on` provides the correct typing already, so you can drop the explicit typing (and end up with more accurate types to boot).

Same with line 224.